components/formats-bsd/src/loci/formats/codec/ZstdCodec.java

```diff
@@ -38,9 +38,22 @@ public byte[] decompress(RandomAccessInputStream in, CodecOptions options)
         catch (EOFException ignored) { }
 
         byte[] data = bytes.toByteArray();
+        return decompress(data);
+    }
+    
+    @Override
+    public byte[] decompress(byte[] data)
+        throws FormatException
```

Comment:
Checking for ``(data == null || data.length == 0)`` to avoid issue
Maybe something like 
```
if (data == null || data.length == 0) {
  throw new IllegalArgumentException("No data to decompress");
}
```